### ISSUE TITLE
Node 6 support

### ISSUE THREAD

**Issue by tcarlsen** [scniro/gulp-clean-css#4]
Im getting this error after the new Node 6 update:

```
path.js:7
    throw new TypeError('Path must be a string. Received ' + inspect(path));
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.dirname (path.js:1324:5)
    at /.../node_modules/clean-css/lib/utils/input-source-map-tracker.js:191:56
    at Array.forEach (native)
    at trackContentSources (/.../node_modules/clean-css/lib/utils/input-source-map-tracker.js:188:20)
    at InputSourceMapStore.trackLoaded (/.../node_modules/clean-css/lib/utils/input-source-map-tracker.js:255:3)
    at fromString (/.../node_modules/clean-css/lib/utils/input-source-map-tracker.js:32:8)
    at InputSourceMapStore.track (/.../node_modules/clean-css/lib/utils/input-source-map-tracker.js:236:5)
    at Object.%
```

**Comment by scniro** [comment]
@tcarlsen hmm, I can see the error is propagating from clean-css as you can see in the node_modules/clean-css folder. Can you let me know which version of this plugin you are using as well as posting your gulp task so I can take a peek? I'd like to get this reproducing on my end. I'll be able to look into this a bit later on today, thanks!

**Comment by tcarlsen** [comment]
`package.json`:
```json
"devDependencies": {
    "babel-preset-es2015": "^6.6.0",
    "gulp": "^3.9.1",
    "gulp-angular-templatecache": "^1.8.0",
    "gulp-autoprefixer": "^3.1.0",
    "gulp-babel": "^6.1.2",
    "gulp-clean-css": "^2.0.6",
    "gulp-concat": "^2.6.0",
    "gulp-jade": "^1.1.0",
    "gulp-ng-annotate": "^2.0.0",
    "gulp-rename": "^1.2.2",
    "gulp-sass": "^2.2.0",
    "gulp-sourcemaps": "^2.0.0-alpha",
    "gulp-stylelint": "^2.0.0",
    "gulp-uglify": "^1.5.3",
    "gulp-xo": "^0.11.0",
    "stylelint-config-standard": "^6.0.0",
    "xo": "^0.15.0"
  },
```

`gulpfile.js`:

```javascript
/* Styles */
gulp.task('styles', () => {
  return gulp.src(`${build}src/css/app.scss`)
    .pipe(sourcemaps.init())
    .pipe(concat('app.css'))
    .pipe(sass())
    .pipe(autoprefixer())
    .pipe(cleancss())
    .pipe(sourcemaps.write('./sourcemaps'))
    .pipe(gulp.dest(build + dest));
});
```

**Comment by bgatellier** [comment]
Same thing here with the shiny new Node 6.

**Comment by scniro** [comment]
@tcarlsen @bgatellier I believed I tracked down the issue. Please pull [2.0.7](https://www.npmjs.com/package/gulp-clean-css) and verify this resolved?

**Comment by samsha1971** [comment]
@scniro upgrade gulp-clean-css to v2.0.7，no error.

**Comment by tcarlsen** [comment]
nice job. THX!

**Comment by sabertazimi** [comment]
Thx.Nice job!

**Comment by turnerhayes** [comment]
I'm still getting this error; I've upgraded to gulp-clean-css@2.0.7

`package.json`:

    "dependencies": {
    "babel-preset-es2015": "^6.6.0",
    "babelify": "^7.2.0",
    "body-parser": "^1.13.3",
    "browserify": "^13.0.0",
    "buffer-reader": "^0.1.0",
    "connect-mongo": "^1.1.0",
    "cookie-parser": "^1.3.5",
    "debug": "^2.2.0",
    "del": "^2.2.0",
    "dotenv": "^2.0.0",
    "express": "~4.13.1",
    "express-hbs": "^1.0.0",
    "express-session": "^1.13.0",
    "gulp": "^3.9.1",
    "gulp-changed": "^1.3.0",
    "gulp-clean-css": "^2.0.7",
    "gulp-debug": "^2.1.2",
    "gulp-if": "^2.0.0",
    "gulp-jshint": "^2.0.0",
    "gulp-less": "^3.0.5",
    "gulp-plumber": "^1.1.0",
    "gulp-rename": "^1.2.2",
    "gulp-sourcemaps": "^2.0.0-alpha",
    "gulp-uglify": "^1.5.3",
    "gulp-util": "^3.0.7",
    "gulp-watch": "^4.3.5",
    "hbs": "~3.1.0",
    "hbsfy": "^2.7.0",
    "jasmid-module": "^1.2.0",
    "jshint": "^2.9.2",
    "jshint-stylish": "^2.1.0",
    "less-plugin-autoprefix": "^1.5.1",
    "less-plugin-clean-css": "^1.5.1",
    "lodash": "^4.11.1",
    "merge-stream": "^1.0.0",
    "mongoose": "^4.4.13",
    "morgan": "^1.6.1",
    "passport": "^0.3.2",
    "passport-facebook": "^2.1.0",
    "passport-google-oauth": "^1.0.0",
    "passport-local": "^1.0.0",
    "q": "^1.4.1",
    "recursive-readdir": "^2.0.0",
    "run-sequence": "^1.1.5",
    "serve-favicon": "~2.3.0",
    "vinyl-buffer": "^1.0.0",
    "vinyl-source-stream": "^1.1.0",
    "watchify": "^3.7.0"
    }

**Comment by scniro** [comment]
@turnerhayes can you post a stack trace and your gulp task? Try deleting `node_modules` and reinstalling for good measure? I fixed this and all seemed to work, so I can't reproduce anything based on what you have shared alone.

**Comment by turnerhayes** [comment]
Stack trace:

    Path must be a string. Received undefined in file undefined line no. undefined TypeError: Path must be a string. Received undefined in file undefined line no. undefined
    at assertPath (path.js:7:11)
    at Object.dirname (path.js:1324:5)
    at /projects/websites/webkb/node_modules/clean-css/lib/utils/input-source-map-tracker.js:191:56
    at Array.forEach (native)
    at trackContentSources (/projects/websites/webkb/node_modules/clean-css/lib/utils/input-source-map-tracker.js:188:20)

    at InputSourceMapStore.trackLoaded (/projects/websites/webkb/node_modules/clean-css/lib/utils/input-source-map-tracker.js:255:3)
    at fromString (/projects/websites/webkb/node_modules/clean-css/lib/utils/input-source-map-tracker.js:32:8)
    at InputSourceMapStore.track (/projects/websites/webkb/node_modules/clean-css/lib/utils/input-source-map-tracker.js:236:5)
    at /projects/websites/webkb/node_modules/clean-css/lib/clean.js:145:44
    at CleanCSS.minify (/projects/websites/webkb/node_modules/clean-css/lib/clean.js:127:42)


Gulp task:


    gulp.task('styles', ['copy-fonts'], function() {
	return gulp.src(path.join(stylesDirectory, 'main.less'))
		.pipe(gulpDebug())
		.pipe(
			plumber(
				function(err) {
					gutil.log(gutil.colors.red('LESS error:\n'), err.message, err.stack || '');
				}
			)
		)
		.pipe(sourcemaps.init())
		.pipe(less({
			modifyVars: {
				"@fa-font-path": '"../fonts"'
			},
			plugins: [cleancss, autoprefix]
		}))
		.pipe(rename({
			dirname: '',
			basename: 'build',
			extname: '.css'
		}))
		.pipe(sourcemaps.write())
		.pipe(gulp.dest(path.join(config.paths.static, 'dist', 'css/')));
    });


The fix linked above seems to work, but doesn't seem to be included in the package when I `npm install` it.

**Comment by scniro** [comment]
@turnerhayes can you explain what you mean by the fix linked above? That change is published to npm. How were you able to verify that?

**Comment by turnerhayes** [comment]
When I manually add the code linked:

    if (typeof(sourceFile) == 'undefined')
       return;

it works, but when I install it from npm it's not there.

**Comment by scniro** [comment]
The fix was actually in the commit seen here, not that code snippet: f62380bcffe1b4b57b0c07927cb1ac9cf50e0f80

Unfortunately I am still struggling to identify the issue. Have you entirely deleted your node modules and tried reinstalling?

**Comment by turnerhayes** [comment]
Ah, ok. Then that fix isn't working, I'm still getting the LESS error. I have deleted node_modules and reinstalled, still nothing.

**Comment by scniro** [comment]
@turnerhayes  Still looking into this and trying to get something reproducing. I know it's not ideal, but have you tried to just `pipe` the plugins after your `less` task? Does the following produce the same result?

.pipe(sourcemaps.init())
    .pipe(less({
        modifyVars: {
            "@fa-font-path": '"../fonts"'
        }
    }))
    .pipe(autoprefix())
    .pipe(cleancss())
    /*...*/

Was your original task working as is before the node upgrade? 6.0.0 or a bump higher (e.g. 6.2.0)?

**Comment by scniro** [comment]
@turnerhayes Looks like an issue was brought up with the underlying clean-css. Lets see where this pr will go jakubpawlowicz/clean-css#774

**Comment by turnerhayes** [comment]
Great, I'll keep an eye on that. Thanks!

**Comment by scniro** [comment]
gulp-clean-css 2.0.8 has been published with the 3.4.13 dep https://github.com/scniro/gulp-clean-css/commit/32909e1eed30a536175afd2aa7d26727877a025a
